Automattic/woocommerce-payments
# feat:  payment methods tooltip info

Fixes https://github.com/Automattic/woocommerce-payments/issues/2474

#### Changes proposed in this Pull Request









I am creating a separate `Tooltip` component, and using it in the checkboxes.
I considered using a separate library for this (https://github.com/Automattic/woocommerce-payments/pull/2483), but then I thought that I might get some push back. So here we are.

The reason for a separate component is that Gutenberg's `Tooltip` component doesn't play nicely when displayed within modals, causing some overflow:

![2021-07-16 17 20 24](https://user-images.githubusercontent.com/273592/126239699-b00101d7-3989-4721-95ca-93d6c0952526.gif)

In order to fix the behavior, I am using a [Portal](https://reactjs.org/docs/portals.html) to render the element in a fixed position outside of the flow.

#### Testing instructions





- Ensure you have the "Enable UPE checkout" flag enabled
- Ensure you have the "Enable UPE settings changes" flag enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Click on the "Add payment method" button
- Hover over the tooltip elements (or click on them ~- PLEASE NOTE: when rendered within a modal, the modal's click handler can still be a bit finnicky, especially with the icon, making the modal close - it's not a problem when rendered outside of a modal and clicking on the button~ EDIT: solved in 949bc2b1 )
- Tooltip should appear
- When clicking, clicking away can also hide the tooltip
- No more than one tooltip will be rendered at one time

![2021-07-19 15 48 14](https://user-images.githubusercontent.com/273592/126239852-fc0154b7-b394-494a-bed0-4728df3ec557.gif)

~Issue when clicking within a modal, causing the modal to close - this is happening because of the modal's own click handler and NOT because of the popover component. Hover works fine:~ EDIT: solved in 949bc2b1
![2021-07-19 21 11 59](https://user-images.githubusercontent.com/273592/126251737-7b71a4c1-2e69-43ef-a48b-0d45fc085fc3.gif)



-------------------
